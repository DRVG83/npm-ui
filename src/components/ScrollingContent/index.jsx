import React from "react";

export default class ScrollingContent extends React.Component {
  render() {
    return (
      <div className={`${this.props.className} panel--scrollingContent`}>
        {this.props.children}
      </div>
    );
  }
}
