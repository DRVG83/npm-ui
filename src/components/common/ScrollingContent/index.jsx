import React from "react";
import styles from "./styles.scss";

export default class ScrollingContent extends React.Component {
  scrollToBottom() {
    this.container.scrollTop = this.container.scrollHeight;
  }
  
  render() {
    return (
      <div
        className={`${this.props.className} ${styles.panelSrollingContent}`}
        ref={(div) => { this.container = div; }}
      >
        {this.props.children}
      </div>
    );
  }
}
