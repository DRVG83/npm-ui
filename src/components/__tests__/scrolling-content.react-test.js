import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import ScrollingContent from "../ScrollingContent";

test("ScrollingContent should render", () => {
  const component = renderer.create(
    <ScrollingContent className="some-class-name">Content</ScrollingContent>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("ScrollingContent should contain the custom className", () => {
  const customClassName = "some-class-name";
  const wrapper = shallow(
    <ScrollingContent className={customClassName}>Content</ScrollingContent>
  );
  expect(wrapper.find("div").hasClass(customClassName)).toBe(true);
});
