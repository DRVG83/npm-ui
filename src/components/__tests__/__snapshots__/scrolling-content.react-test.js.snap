// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingContent should render 1`] = `
<div
  className="some-class-name panel--scrollingContent"
>
  Content
</div>
`;
